### DISCUSSION TITLE
A New Chapter for Nuxt: Seamless Integration of Nuxt with Backend Frameworks!

### DISCUSSION THREAD

**Discussion by trylovetom** [nuxt/nuxt#24249] [Sharing]
Hey Guys! I'm thrilled to share a method I've developed for integrating other backend frameworks with Nuxt. This approach enables hot reload and devtool usage in the development environment, and it's deployable in production settings. I've already tried out this technique by creating a small package and example using Bun and elysiajs. You can check it out here: https://github.com/trylovetom/elysiajs-nuxt

https://github.com/nuxt/nuxt/assets/13268073/d5c96902-698f-482a-8360-9f657ab40eba

The logic in the development environment mainly involves hijacking the script path of the Nuxt Vite dev server, inspired by https://vitejs.dev/guide/backend-integration.html. For the production environment, it's primarily achieved by customizing Nitro's presets.

I believe that if the Vite dev server can integrate with many backend frameworks, Nuxt definitely can too! I'm really excited about this and deeply admire the Nuxt community and team for their innovative spirit. I've always loved Nuxt for its versatility and efficiency, and I hope my contribution can add value to our fantastic community.

If you have any suggestions or questions, feel free to raise them for discussion! I'm looking forward to growing with the community and continuously improving this innovative solution.

**Comment by hunteryun** [comment]
PHP Backend possible？
